src/diffusers/schedulers/scheduling_utils.py

```diff
@@ -37,6 +38,28 @@ class SchedulerOutput(BaseOutput):
     prev_sample: torch.FloatTensor
 
 
+class BaseScheduler(abc.ABC):
+
+    def scale_initial_noise(self, noise: torch.FloatTensor):
+        """
+        Scales the initial noise to the correct range for the scheduler.
+        """
+        return noise
+
+    def scale_model_input(self, sample: torch.FloatTensor, step: int):
+        """
+        Scales the model input (`sample`) to the correct range for the scheduler.
+        """
+        return sample
+
+    @abc.abstractmethod
+    def get_noise_condition(self, step: int):
+        """
+        Returns the input noise condition for the model (e.g. `timestep` or `sigma`).
+        """
+        raise NotImplementedError("Scheduler must implement the `get_noise_condition` function.")
+
+
```

Comment:
This class combines the new required functions and ideally should be merged with `SchedulerMixin` (left standalone for easier reviewing for now).

Reply:
Yes, I agree to have it merged; otherwise schedulers need to inherit from both.